doc/reference/internals/iproto/replication.rst

```diff
@@ -135,6 +163,70 @@ is an optional key used in the SUBSCRIBE request followed by an array
 of ids of instances whose rows won't be relayed to the replica.
 The field is encoded only when the ID list is not empty.
 
+..  _box_protocol-fetch-snapshot:
+
+IPROTO_FETCH_SNAPSHOT
+~~~~~~~~~~~~~~~~~~~~~
+
+Code: 0x45.
+
+To join a replica set as an anonymous replica, an instance must send an initial
+IPROTO_FETCH_SNAPSHOT request to any node in the replica set:
+
+..  raw:: html
+    :file: images/repl_fetch_snapshot_request.svg
+
+To learn about anonymous replicas, see :ref:`replication.anon <configuration_reference_replication_anon>`.
+
+..  include:: replication.rst
+    :start-after: iproto_join_response_sequence_start
+    :end-before: iproto_join_response_sequence_end
+
+..  _box_protocol-register:
+
+IPROTO_REGISTER
+~~~~~~~~~~~~~~~
+
+Code: 0x46.
+
+To register an anonymous replica in a replica set so that it's not anonymous anymore,
+it must send an IPROTO_REGISTER request to any (TODO: or master?) node in the replica set:
+
+TODO: request structure diagram
```

Comment:
@bitgorbovsky @sergepetrenko please assist with TODOs:
- which instances can accept IPROTO_REGISTER?
- what is the request structure?

Reply:
The request should be sent to master (like JOIN).

REGISTER contains
MP_STR instance uuid of the sender
(optional) MP_STR instance name (this only applies to 3.0 docs)
MP_MAP vclock of the instance to be registered.

A response to REGISTER contains everything from a JOIN response that is not covered by FETCH_SNAPSHOT response

Basically, IPROTO_FETCH_SNAPSHOT + IPROTO_REGISTER are equivalent to IPROTO_JOIN.

IPROTO_FETCH_SNAPSHOT performs the "initial join" stage, where a stream of snapshot rows is sent to a replica, and IPROTO_REGISTER performs the "final join" stage, where the node is registered, and a follow-up stream of xlog contents is fed to the replica (until it receives a row with its own registration).